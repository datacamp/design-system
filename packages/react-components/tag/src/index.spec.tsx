import '@testing-library/jest-dom/extend-expect';

import axeRender from '@datacamp/waffles-axe-render';
import { getColor, getContrastColor } from '@datacamp/waffles-core';
import { matchers } from 'jest-emotion';
import React from 'react';

import Tag from '.';

expect.extend(matchers);

const exampleTagBody = '50XP';

describe('<Tag />', () => {
  it('creates a default tag', async () => {
    const { container } = await axeRender(<Tag>{exampleTagBody}</Tag>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('sets the background color and infers text color', async () => {
    const { getByText } = await axeRender(
      <Tag color="red">{exampleTagBody}</Tag>
    );
    const backgroundColor = getColor('red');
    const textColor = getContrastColor('red');

    expect(getByText(exampleTagBody)).toHaveStyleRule(
      'background-color',
      backgroundColor
    );
    expect(getByText(exampleTagBody)).toHaveStyleRule('color', textColor);
  });

  it('sets the text color based on the property when provided', async () => {
    const { getByText } = await axeRender(
      <Tag textColor="red">{exampleTagBody}</Tag>
    );

    expect(getByText(exampleTagBody)).toHaveStyleRule('color', getColor('red'));
  });

  it('has a default border radius', async () => {
    const { getByText } = await axeRender(<Tag>{exampleTagBody}</Tag>);

    expect(getByText(exampleTagBody)).toHaveStyleRule('border-radius', '12px');
  });

  it('sets the border radius when rounded=true', async () => {
    const { getByText } = await axeRender(<Tag rounded>{exampleTagBody}</Tag>);

    expect(getByText(exampleTagBody)).toHaveStyleRule('border-radius', '4px');
  });

  it('Adds the extraClass to the className', async () => {
    const { getByText } = await axeRender(
      <Tag extraClass="test-class">{exampleTagBody}</Tag>
    );

    expect(getByText(exampleTagBody)).toHaveClass('test-class');
  });
});
