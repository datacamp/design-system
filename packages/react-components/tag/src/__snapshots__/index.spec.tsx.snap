// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> creates a default tag 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #e6eaeb;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 0.75rem;
  font-weight: 700;
  font-family: Lato;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  line-height: 1;
  padding: 5px 11px;
  position: relative;
  text-transform: uppercase;
  white-space: nowrap;
  background-color: #ebf4f7;
  border-radius: 12px;
}

<Tag>
  <StyledTagDiv>
    <div
      className="emotion-0 emotion-1"
    >
      50XP
    </div>
  </StyledTagDiv>
</Tag>
`;
